internal/api/mfa.go

```diff
@@ -90,10 +90,27 @@ func (a *API) enrollPhoneFactor(w http.ResponseWriter, r *http.Request, params *
 	if err := models.DeleteExpiredFactors(db, config.MFA.FactorExpiryDuration); err != nil {
 		return err
 	}
-
-	for _, factor := range factors {
-		if factor.IsVerified() {
-			numVerifiedFactors += 1
+	for _, factor := range user.Factors {
+		switch {
+		case factor.FriendlyName == params.FriendlyName:
+			return unprocessableEntityError(
+				ErrorCodeMFAFactorNameConflict,
+				fmt.Sprintf("A factor with the friendly name %q for this user already exists", factor.FriendlyName),
+			)
+
+		case factor.IsVerified() && factor.IsPhoneFactor():
```

Comment:
Add two test for these two cases:

1. When there's an existing verified phone factor we should receive an error
2. When there's an existing unverified phone factor all existing unverified phone factors should be removed